### PR TITLE
Relationship

### PR DESCRIPTION
**Pull request by kemigabocatherine**
### Changes made

- Created new tables called specializations, vets and visits
- Inserted data into the specializations, visits, and vets tables
- Wrote queries to answer questions about relationships of the tables

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -93,3 +93,92 @@ UPDATE animals
...
+INSERT INTO specializations(vet_id, species_id)
+VALUES(3, 2);
+
+INSERT INTO specializations(vet_id, species_id)
+VALUES(4, 2);
+
+/* visits data */
+
+INSERT INTO visits(animal_id, vet_id, visit_date)
+vet_clinic-# VALUES(1, 1, 'May 24, 2020');
```
Comment (rindrajosia) [inline data.sql]: - [ ] [REQUIRED] Please check this line as, I got an error when I run it

File: data.sql [hunk 2]
```diff
@@ -93,3 +93,92 @@ UPDATE animals
 SET owner_id = 5
 WHERE name = 'Angemon' OR name = 'Boarmon';
 
+/* Vets data */
+
+INSERT INTO vets(name, age, date_of_graduation)
+vet_clinic-# VALUES('William Tatcher', 45, 'Apr 23, 2000');
```
Comment (rindrajosia) [inline data.sql:99]: - [ ] [REQUIRED] Please check this line as, I got an error when I run it

File: queries.sql [hunk 3]
```diff
@@ -117,3 +117,67 @@ GROUP BY owners.full_name
...
+SELECT COUNT(*) FROM Visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.ID
+JOIN specializations ON vets.id = specializations.vet_id
+JOIN species ON specializations.species_id = species.id
+WHERE animals.species_id NOT IN (
+  SELECT specializations.species_id FROM vets
+JOIN specializations ON vets.id = specializations.vet_id
+WHERE vets.id = visits.vet_id
+);
```
Comment (rindrajosia) [inline queries.sql]: - [ ] [REQUIRED] Please make sure that this query return correct data. Your query return 2 however, it should be 11.

File: queries.sql [hunk 4]
```diff
@@ -117,3 +117,67 @@ GROUP BY owners.full_name
...
+
+/* What specialty should Maisy Smith consider getting? Look for the species she gets the most. */
+SELECT species.name, COUNT(species.name) FROM vets
+JOIN visits ON vets.id = visits.vet_id
+JOIN animals ON visits.animal_id = animals.id
+JOIN species ON animals.species_id = species.id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY species.name
+LIMIT 1;
```
Comment (rindrajosia) [inline queries.sql]: - [ ] [REQUIRED] This query should return 2  rows, please remove **LIMIT 1** because it force your query to return only 1 row.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:

## To highlights 💪 

- most of your queries provides correct data

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please, add querie that  List all vets and their specialties, including vets with no specialties. It seems that you forget it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (kemigabocatherine) [reply queries.sql]: Thank you for your helpful feedback.
I got my data here as 12 and I checked it manually, it was also 12.

File: data.sql (hunk 2 above)
Comment (rindrajosia) [reply data.sql:99]: - [ ] [REQUIRED] It will be nice if you fix this one also ⬆️

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ##To hiqhlights 💪 

- Good job by working on the last requirements.

 ## Required Changes ♻️

- [ ] **_Please, add querie that List all vets and their specialties, including vets with no specialties. It seems that you forget it._**

-Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by keneogu — APPROVED** [review]
> ## Hi @kemigabocatherine,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> ## Congratulations!     
> Cheers and Happy coding!clapclapclap
> 
> image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966) image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966)
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**